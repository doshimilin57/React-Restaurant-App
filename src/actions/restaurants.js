import axios from 'axios';
import { SET_RESTAURANTS, SHOW_MORE_RESTAURANTS, SET_PAGE } from './types';
import { setRestaurantsCount } from './results';

// SET_RESTAURANTS
export const setRestaurants = (restaurants) => ({
  type: SET_RESTAURANTS,
  restaurants,
});

// SHOW_MORE_RESTAURANTS
export const setShowMoreRestaurants = (restaurants) => ({
  type: SHOW_MORE_RESTAURANTS,
  restaurants,
});

// SHOW_MORE_RESTAURANTS
export const setPage = (page) => ({
  type: SET_PAGE,
  page,
});

export const startSearchRestaurants = (city = '') => {
  return async (dispatch) => {
    try {
      const res = await axios.get(
        `http://opentable.herokuapp.com/api/restaurants?city=${city}&per_page=25&page=1`
      );
      dispatch(setRestaurants(res.data.restaurants));
      dispatch(setRestaurantsCount(res.data.total_entries));
    } catch (e) {
      console.log(e, e.response);
    }
  };
};

export const startShowMoreRestaurants = (city = '') => {
  return async (dispatch, getState) => {
    try {
      const page = getState().results.page || 1;
      const res = await axios.get(
        `http://opentable.herokuapp.com/api/restaurants?city=${city}&per_page=25&page=${
          page + 1
        }`
      );
      dispatch(setShowMoreRestaurants(res.data.restaurants));
      dispatch(setPage(page + 1));
    } catch (e) {
      console.log(e, e.response);
    }
  };
};
