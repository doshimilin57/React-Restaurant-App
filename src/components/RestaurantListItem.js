import React from 'react';
import numeral from 'numeral';

const RestaurantListItem = ({
  id,
  name,
  price,
  address,
  city,
  state,
  country,
  postal_code,
}) => (
  <div className="list-item">
    <div>
      <h3 className="list-item__title">{name}</h3>
      <span className="list-item__sub-title">
        {address}
        {', '}
        {city}
        {', '}
        {state}
        {', '}
        {country}
        {', '}
        {postal_code}
      </span>
    </div>
    <h3 className="list-item__data">{numeral(price).format('$0,0.00')}</h3>
  </div>
);

export default RestaurantListItem;
