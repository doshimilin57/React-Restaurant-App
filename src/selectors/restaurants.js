import moment from 'moment';

// Get visible restaurants

export default (restaurants, { text }) => {
  return restaurants.filter((restaurant) => {
    const textMatch = restaurant.name
      .toLowerCase()
      .includes(text.toLowerCase());
    const addressMatch = restaurant.address
      .toLowerCase()
      .includes(text.toLowerCase());
    const areaMatch = restaurant.area
      .toLowerCase()
      .includes(text.toLowerCase());
    const cityMatch = restaurant.city
      .toLowerCase()
      .includes(text.toLowerCase());
    const stateMatch = restaurant.state
      .toLowerCase()
      .includes(text.toLowerCase());
    const postalCodeMatch = restaurant.postal_code
      .toLowerCase()
      .includes(text.toLowerCase());
    return (
      textMatch ||
      addressMatch ||
      areaMatch ||
      cityMatch ||
      stateMatch ||
      postalCodeMatch
    );
  });
};
