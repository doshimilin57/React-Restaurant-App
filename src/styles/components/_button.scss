.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
  height: 100%;
}

.button--link {
  background: none;
  color: #888;
  font-size: 1.2rem;
}

.button--secondary {
  background: #888;
}
