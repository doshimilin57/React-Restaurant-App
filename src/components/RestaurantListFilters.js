import React from "react";
import { connect } from "react-redux";
import { setTextFilter } from "../actions/filters";
import selectRestaurants from "../selectors/restaurants";
import {
  startSearchRestaurants,
  startShowMoreRestaurants,
} from "../actions/restaurants";

export class RestaurantListFilters extends React.Component {
  state = {
    city: "",
    displayRefineBox: false,
  };

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  onCityChange = (e) => {
    const city = e.target.value;
    this.setState(() => ({ city }));
  };

  submitHandler = (e) => {
    e.preventDefault();
    this.props.startSearchRestaurants(this.state.city);
  };

  startShowMoreRestaurants = (e) => {
    e.preventDefault();
    this.props.startShowMoreRestaurants(this.state.city);
  };

  render() {
    return (
      <div className="content-container list-filters">
        <form onSubmit={this.submitHandler}>
          <div className="input-group">
            <div className="input-group__item">
              <input
                type="text"
                className="text-input"
                placeholder="Enter City"
                aria-labelledby="lblCity"
                tabindex="-1"
                value={this.state.city}
                onChange={this.onCityChange}
              />
            </div>
            <div className="input-group__item">
              <input
                type="text"
                className="text-input"
                // placeholder="Filter(Name, Address, Area)"
                placeholder="Filter Data"
                aria-labelledby="lblFilter"
                tabindex="-1"
                value={this.props.filters.text}
                onChange={this.onTextChange}
              />
            </div>
            <div>
              <button type="submit" className="button">
                Search
              </button>
            </div>
            <div>
              <button
                type="click"
                className="button button--link"
                onClick={this.startShowMoreRestaurants}
                disabled={
                  this.props.restaurantCount >= this.props.total_entries
                }
              >
                Show More
              </button>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const visibleRestaurants = selectRestaurants(
    state.restaurants,
    state.filters
  );
  return {
    filters: state.filters,
    total_entries: state.results.total_entries,
    restaurantCount: visibleRestaurants.length,
  };
};

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  startSearchRestaurants: (city) => dispatch(startSearchRestaurants(city)),
  startShowMoreRestaurants: (city) => dispatch(startShowMoreRestaurants(city)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RestaurantListFilters);
