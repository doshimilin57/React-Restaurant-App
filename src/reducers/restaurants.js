// Restaurants Reducer

import { SET_RESTAURANTS, SHOW_MORE_RESTAURANTS } from '../actions/types';

const restaurantsReducerDefaultState = [];

export default (state = restaurantsReducerDefaultState, action) => {
  switch (action.type) {
    case SET_RESTAURANTS:
      return action.restaurants;
    case SHOW_MORE_RESTAURANTS:
      return [...state, ...action.restaurants];
    default:
      return state;
  }
};
