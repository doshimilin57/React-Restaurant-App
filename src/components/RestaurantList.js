import React from 'react';
import { connect } from 'react-redux';
import RestaurantListItem from './RestaurantListItem';
import selectRestaurants from '../selectors/restaurants';

export const RestaurantList = ({ restaurants }) => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Restaurants</div>
      <div className="show-for-desktop">Restaurant</div>
      <div className="show-for-desktop">Price</div>
    </div>
    <div className="list-body">
      {restaurants.length === 0 ? (
        <div className="list-item list-item--message">
          <span>No results!</span>
        </div>
      ) : (
        restaurants.map((restaurant) => {
          return <RestaurantListItem key={restaurant.id} {...restaurant} />;
        })
      )}
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    restaurants: selectRestaurants(state.restaurants, state.filters),
  };
};

export default connect(mapStateToProps)(RestaurantList);
