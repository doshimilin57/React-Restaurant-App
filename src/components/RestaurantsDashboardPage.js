import React from "react";
import RestaurantList from "./RestaurantList";
import RestaurantListFilters from "./RestaurantListFilters";

const RestaurantsDashboardPage = () => (
  <div>
    <RestaurantListFilters />
    <RestaurantList />
  </div>
);

export default RestaurantsDashboardPage;
