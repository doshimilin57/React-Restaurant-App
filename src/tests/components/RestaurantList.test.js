import React from 'react';
import { shallow } from 'enzyme';
import { RestaurantList } from '../../components/RestaurantList';
import restaurants from '../fixtures/restaurants';

test('should render restaurants list with restaurants', () => {
  const wrapper = shallow(<RestaurantList restaurants={restaurants} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render restaurants list with empty array', () => {
  const wrapper = shallow(<RestaurantList restaurants={[]} />);
  expect(wrapper).toMatchSnapshot();
});
