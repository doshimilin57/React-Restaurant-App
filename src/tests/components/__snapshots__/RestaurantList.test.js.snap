// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render restaurants list with empty array 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Restaurants
    </div>
    <div
      className="show-for-desktop"
    >
      Restaurant
    </div>
    <div
      className="show-for-desktop"
    >
      Price
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item--message"
    >
      <span>
        No results!
      </span>
    </div>
  </div>
</div>
`;

exports[`should render restaurants list with restaurants 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Restaurants
    </div>
    <div
      className="show-for-desktop"
    >
      Restaurant
    </div>
    <div
      className="show-for-desktop"
    >
      Price
    </div>
  </div>
  <div
    className="list-body"
  >
    <RestaurantListItem
      address="169 Niagara Street"
      area="Toronto / SW Ontario"
      city="Toronto"
      country="CA"
      id={91948}
      image_url="https://www.opentable.com/img/restimages/91948.jpg"
      lat={43.641939}
      lng={-79.406543}
      mobile_reserve_url="http://mobile.opentable.com/opentable/?restId=91948"
      name="Edulis Restaurant"
      phone="4167034222"
      postal_code="M5V 1C9"
      price={3}
      reserve_url="http://www.opentable.com/single.aspx?rid=91948"
      state="ON"
    />
    <RestaurantListItem
      address="655 Dixon Rd."
      area="Toronto / SW Ontario"
      city="Toronto"
      country="CA"
      id={44791}
      image_url="https://www.opentable.com/img/restimages/44791.jpg"
      lat={43.6898}
      lng={-79.578}
      mobile_reserve_url="http://mobile.opentable.com/opentable/?restId=44791"
      name="Ginko Japanese Restaurant"
      phone="4162488445"
      postal_code="M5H 3M7"
      price={3}
      reserve_url="http://www.opentable.com/single.aspx?rid=44791"
      state="ON"
    />
    <RestaurantListItem
      address="1585 Dundas West"
      area="Toronto / SW Ontario"
      city="Toronto"
      country="CA"
      id={75154}
      image_url="https://www.opentable.com/img/restimages/75154.jpg"
      lat={43.649767}
      lng={-79.434577}
      mobile_reserve_url="http://mobile.opentable.com/opentable/?restId=75154"
      name="Lula Lounge"
      phone="4165880307"
      postal_code="M6K 1T5"
      price={3}
      reserve_url="http://www.opentable.com/single.aspx?rid=75154"
      state="ON"
    />
    <RestaurantListItem
      address="214 King St. West"
      area="Toronto / SW Ontario"
      city="Toronto"
      country="CA"
      id={14806}
      image_url="https://www.opentable.com/img/restimages/14806.jpg"
      lat={43.647429}
      lng={-79.386774}
      mobile_reserve_url="http://mobile.opentable.com/opentable/?restId=14806"
      name="Il Fornello on King"
      phone="4169772855"
      postal_code="M5H 3S6"
      price={3}
      reserve_url="http://www.opentable.com/single.aspx?rid=14806"
      state="ON"
    />
    <RestaurantListItem
      address="190 University Ave"
      area="Toronto / SW Ontario"
      city="Toronto"
      country="CA"
      id={150661}
      image_url="https://www.opentable.com/img/restimages/150661.jpg"
      lat={43.649368}
      lng={-79.385921}
      mobile_reserve_url="http://mobile.opentable.com/opentable/?restId=150661"
      name="Momofuku DAISHÅŒ - Toronto"
      phone="6472538000"
      postal_code="M5H 0A3"
      price={4}
      reserve_url="http://www.opentable.com/single.aspx?rid=150661"
      state="ON"
    />
  </div>
</div>
`;
